salt/metalk8s/addons/ui/deployed/ui-configuration.sls

```diff
@@ -67,9 +67,41 @@ Create metalk8s-ui-config ConfigMap:
 
 {%- else %}
 
+  {%- if metalk8s_shell_ui_config is not none %}
+    {%- set config_data = metalk8s_shell_ui_config.data['config.yaml'] | load_yaml %}
+
+    {%- if config_data.apiVersion == 'addons.metalk8s.scality.com/v1alpha1' and 'oidc' in config_data.spec %}
+
+Create metalk8s-ui-config ConfigMap:
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: metalk8s-ui-config
+          namespace: metalk8s-ui
+        data:
+          config.yaml: |-
+            apiVersion: addons.metalk8s.scality.com/v1alpha2
+            kind: UIConfig
+            spec:
+              kind: "OIDC"
+              {{ config_data.spec.oidc | yaml(False) | indent(14) }}
+
+    {%- else %}
+
+metalk8s-ui-config ConfigMap already exist:
+  test.succeed_without_changes: []
+
+    {%- endif %}
+
+  {%- else %}
+
```

Comment:
Not sure to understand this part:

- we have both `metalk8s-ui-config` and `metalk8s-shell-ui-config` already existing
- we want to copy over the OIDC settings from the latter to the former, in case the `shell-ui` config is in the old format

But if that's true, we're overwriting the previous `metalk8s-ui-config`, without checking its own apiVersion or previous contents (what if we changed the basePath?)